src/index.js

```diff
@@ -6,46 +6,55 @@ const tags = tagsW3.map(tag => tag.replace(/[<|>]/g, ""));
 
 // Wait for the DOM to be fully loaded
 document.addEventListener("DOMContentLoaded", function () {
-    uploadTagsFromLocalStorage(); // list all tags in localStorage
-    updateProgressBar();
-    // Find the form element and add a submit event listener
-    const form = document.querySelector("form");
-    form.addEventListener("submit", function (event) {
-        // on submit
-        event.preventDefault(); // Prevent the default form submission
-        submitVal(); // call our submitVal function
-    });
 
-    function submitVal() {
-        let val = document.getElementById("input1").value;
-        // remove all white spaces and non-alphabetical characters except . ! - 
-        let regex = /[^a-zA-Z!-.]/g;
-        // change input to lower case and apply regex
-        let sanitizedVal = val.toLowerCase().replace(regex, "");
+  uploadTagsFromLocalStorage(); // list all tags in localStorage
+  updateProgressBar();
+  // Find the form element and add a submit event listener
+  const form = document.querySelector("form");
+  form.addEventListener("submit", function (event) {
+    // on submit
+    event.preventDefault(); // Prevent the default form submission
+    submitVal(); // call our submitVal function
+  });
+
+  function submitVal() {
+    let val = document.getElementById("input1").value;
+    // remove all white spaces and non-alphabetical characters except . ! - 
+    let regex = /[^a-zA-Z!-.]/g;
+    // change input to lower case and apply regex
+    let sanitizedVal = val.toLowerCase().replace(regex, "");
+    let li = "<li>" + sanitizedVal + "</li>";
+    // check if the input value is in the tags array
+    let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
+    
+    // get the list of inputs already made
+    const getCurrentList = document.querySelectorAll("#list li");
+    let checkDuplicate = false;
+    // check if the input is already in the list
+    getCurrentList.forEach(li => {
+      if (li.textContent.toLowerCase() === sanitizedVal) {
+      checkDuplicate = true;
+      }
+    });
```

Comment:
This is very clean and logical, and given that an array of max 124 length is small by today's standards, from a performance point of view will work fine. It's true that directly querying the DOM is slower than traversing an array, and includes() will be theoretically more performant.

Hint: in line 31, instead of querying the DOM, grab the array from local storage. Then in line 40, the second condition should have array.includes(...)

Reply:
Another suggestion: use a `Set`. This is a special object with a `.has(...)` property. A `Set` is implemented in a very efficient way, so checking for the existence of a value does not involve traversing the entire Set. The time complexity would be O(1).

Reply:
This is what I was trying to do, by writing `const getCurrentList = localStorage.getItem("myTags");` which does show the array in `console.log` when I write it outside the `eventlistener` but not inside it on `line 31` to work with it.

So, after changing `if (found && !checkDuplicate)` to `if (found && !getCurrentList.includes(found))`, the function does not work anymore.

Reply:
I found the issue I was facing when using `includes()`, so I am one step closer to what I want to achieve 💟.

**Problems I have to take care of:**

- Create the same functionality with using `.has()`. It worked with some tags, but not with everyone (when the tag was one letter and it was in the list, the state was `true`, when it has more than 1 letter, the state was `false` and it was added to the list)
- When there is no value added (at the beginning of the quiz), `localStorage.getItem("myTags")` output is `null`, which throws a `Reference Error` and cannot be worked with. As soon as a value is set, `includes()` is working.

I can do it 😎